tests/foreman/ui/test_remoteexecution.py

```diff
@@ -823,3 +827,279 @@ def test_positive_run_job_against_multiple_provisioned_hosts(self):
+                status = self.job.run(
+                    job_category='Miscellaneous',
+                    job_template=jobs_template_name,
+                    options_list=[{'name': 'command', 'value': 'ls'}]
+                )
+                # get job invocation id from the current url
+                invocation_id = self.browser.current_url.rsplit('/', 1)[-1]
+                self.assertTrue(
+                        status,
+                        'host output: {0}'.format(
+                            ' '.join(JobInvocation.get_output({
+                                'id': invocation_id,
+                                'host': client.hostname})
+                            )
+                        )
+                    )
+
+    @upgrade
+    @tier3
+    def test_positive_run_job_template_multiple_hosts_by_ip(self):
+        """Run a job template against multiple hosts by ip
+
+        :id: c4439ec0-bb80-47f6-bc31-fa7193bfbeeb
+
+        :Setup: Create a working job template.
+
+        :Steps:
+
+            1. Set remote_execution_connect_by_ip on hosts to true
+            2. Navigate to the hosts page and select at least two hosts
+            3. Click the "Select Action"
+            4. Select the job and appropriate template
+            5. Run the job
+
+        :expectedresults: Verify the job was successfully ran against the hosts
+
+        :CaseLevel: System
+        """
+        prov_server = settings.compute_resources.libvirt_hostname
+        with VirtualMachine(
+              distro=DISTRO_RHEL7,
+              provisioning_server=prov_server,
+              bridge=settings.vlan_networking.bridge,
+              ) as client:
+            with VirtualMachine(
+                  distro=DISTRO_RHEL7,
+                  provisioning_server=prov_server,
+                  bridge=settings.vlan_networking.bridge,
+                  ) as client2:
+                for vm in client, client2:
+                    vm.install_katello_ca()
+                    vm.register_contenthost(
+                        self.organization.label, lce='Library')
+                    self.assertTrue(vm.subscribed)
+                    add_remote_execution_ssh_key(vm.ip_addr)
+                    Host.update({
+                        'name': vm.hostname,
+                        'subnet-id': self.new_sub['id'],
+                    })
+                    # connect to host by ip
+                    Host.set_parameter({
+                        'host': vm.hostname,
+                        'name': 'remote_execution_connect_by_ip',
+                        'value': 'True',
+                    })
+                with Session(self.browser) as session:
+                    set_context(session, org=self.organization.name)
+                    self.hosts.update_host_bulkactions(
+                        [client.hostname, client2.hostname],
+                        action='Run Job',
```

Comment:
This seem to be outdated - it currently fails with:
'element <|strategy=xpath|value=//li/a[text()='Run Job']|> was not found while trying to click'
the actual text is now "schedule remote job":
![img](http://storage1.static.itmages.com/i/17/1206/h_1512569029_2705250_bbc16ff2cf.png)

shall this be addressed in different issue&&pr?

Reply:
so how does test pass?

Reply:
oh, we access element by id, so probably it was not changed

Reply:
```
                if action == 'Run Job':
                    self.assign_value(
                        locators['job_invocation.command'],
                        parameter['command']
                    )
```

 locators['job_invocation.command'] = "job_invocation.command": (By.XPATH, "//textarea[@id='command']"),

where did you find `'element <|strategy=xpath|value=//li/a[text()='Run Job']|>`?

Actually, I can't find such locator in any file

Reply:
@oshtaier,
I think it's hidden rather here: (few lines above in `update_host_bulkactions`):
```python
268: self.click(locators['host.select_action'] % action)
```
which boils down to (`locators/base.py`):
```python
268: "host.select_action": (By.XPATH, "//li/a[text()='%s']")
```


> so how does test pass?

this is a million dollar question - @pondrejk are you sure you didn't use `satellite 6.2` or something?
anyway, I spot this error in the test which uses the hostname in the current automation. I just assume the `_by_ip` equivalent does all the same